java11/src/main/java/com/fullcontact/apilib/enrich/FullContact.java

```diff
@@ -214,45 +221,163 @@ private HttpClient configureHttpClient() {
    *
    * @param companyRequest original request sent by client
    * @return completed CompletableFuture with CompanySearchResponseList
-   * @throws FullContactException exception if client is shutdown or request validation
+   * @throws FullContactException exception if client is shutdown or request fails validation
    * @see <a href =
    *     "https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html">CompletableFuture</a>
    */
-  public CompletableFuture<CompanySearchResponseList> search(CompanyRequest companyRequest)
+  public CompletableFuture<CompanySearchResponseList> search(
+      CompanyRequest companyRequest, RetryHandler retryHandler) throws FullContactException {
+    checkForShutdown();
+    companyRequest.validateForSearch();
+
+    CompletableFuture<HttpResponse<String>> responseCF = new CompletableFuture<>();
+    HttpRequest httpRequest =
+        this.buildHttpRequest(FCConstants.companySearchUri, gson.toJson(companyRequest));
+    sendRequest(httpRequest, retryHandler, responseCF);
+    return responseCF.thenApply(FullContact::getCompanySearchResponseList);
+  }
+
+  /**
+   * Method for Resolve Identity Map. It converts the request to json, send the Asynchronous request
+   * using HTTP POST method. It also handles retries based on retryHandler specified at FullContact
+   * Client level.
+   *
+   * @param personRequest original request sent by client
+   * @return completed CompletableFuture with ResolveResponse
+   * @throws FullContactException exception if client is shutdown or request fails validation
+   * @see <a href =
+   *     "https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html">CompletableFuture</a>
+   */
+  public CompletableFuture<ResolveResponse> identityMap(PersonRequest personRequest)
```

Comment:
The Resolve requests will look just a bit different than the `PersonRequest`s. @lguegan can you help @prakhar-j  with this if he has questions?  
A couple important differences:
- The Resolve request doesn't take a webhook
- The Resolve request for mapping can't take a person ID
- The Resolve request for mapping must take a field other than record id (email, phone, etc) 

It might be worth having separate objects for Map, Resolve, and Delete Resolve requests

Reply:
I though about this whether to have a separate request class for resolve or use same `PersonRequest`, the differences were less, so decided to go with same `PersonRequest` type for simplicity to user. Also if in future, we decide to add these fields to Resolve then there will be unnecessary duplication. 
To handle differences, 
- I have created a `@GsonExclude` annotation which will ignore the fields not accepted by resolve, I will add it to `webhookUrl` also.
- For mapping, there is a validation check `validateForIdentityMap()`
- I can add a validation for 'must have a field other than recordID' but our API response does handle that. I guess its better to add validation here itself?

Let me know your thoughts on whether we should have separate objects for resolve requests.

Reply:
@nathan-pr what do you think about this? should we have a separate model for resolve requests or the `PersonRequest` will do?

Reply:
One one hand this seems reasonable for now -- the `PersonRequest` is a MultiField Query.  Perhaps we have two -- a `MultiField Request` and then a PersonRequest which extends `MultifieldRequest` adding person specific things like confidence, webhookUrl, and infer flag. I like the idea of having identityMap specific validation though!

Reply:
This was getting little complicated with builder and serialization, so created a separate `ResolveRequest` model. Also added validation for `identity.map`